logprep/framework/pipeline.py

```diff
@@ -191,6 +192,10 @@ def _create_connectors(self):
 
         self._input.setup()
         self._output.setup()
+        if hasattr(self._input, "server"):
+            while self._input.server.config.port in self._used_server_ports:
+                self._input.server.config.port += 1
+            self._used_server_ports.append(self._input.server.config.port)
```

Comment:
Does it actually work to just change the server port like this? I also haven't seen any tests for that if I'm not mistaken. Additionally if it does work I think some notification for the user would be helpful, e.g. a logger writing to stdout that the port was changed to `xxxx`. 
And also are ports ever removed from this list again if a pipeline dies?

Reply:
you discovered the tests already ;-)
yes this works and is the solution for now. We should go with it and check in further tests, if this is a good way to deal with it.
the server process posts the port in the logs and in documentation this behavior is described.